client/src/app/components/Toolbar/components/UserBar/UserBar.jsx

```diff
@@ -9,45 +9,75 @@ import { UserPropType } from 'common/constants/propTypes';
 import SettingsIcon from 'assets/images/cog-solid.svg';
 import UserIcon from 'assets/images/user-solid.svg';
 import LogoutIcon from 'assets/images/sign-out.svg';
+import Overlay, { OverlayStyleType } from 'common/components/Overlay';
 
 class UserBar extends Component {
   state = {
-    hideMenu: true
+    isVisible: false
+  };
+
+  componentDidMount() {
+    document.addEventListener('keydown', this.escapeListener);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.escapeListener);
+  }
+
+  escapeListener = event => {
+    const { code } = event;
+    if (code === 'Escape') {
+      this.setState({
+        isVisible: false
+      });
+    }
   };
 
   handleLogOut = () => {
     const { logoutCurrentUser } = this.props;
     logoutCurrentUser();
   };
 
-  handleShowMenu = () => {
-    const { hideMenu } = this.state;
-    this.setState({ hideMenu: !hideMenu });
+  toggleMenu = () => {
```

Comment:
this could be one line:
this.SetState(({ isVisible: oldValue ) => ({ isVisible: !oldValue }));